server/controllers/finance/accounts/index.js

```diff
@@ -208,22 +209,85 @@ function lookupAccount(id) {
       a.classe, a.is_asset, a.reference_id, a.is_brut_link, a.is_charge,
       a.number, a.label, a.parent, a.type_id, a.is_title, at.type
     FROM account AS a JOIN account_type AS at ON a.type_id = at.id
-    WHERE a.id = ?;
-  `;
+    `;
+
+  sql += id ? ' WHERE a.id = ? ORDER BY CAST(a.number AS CHAR(15)) ASC;' : ' ORDER BY CAST(a.number AS CHAR(15)) ASC;';
```

Comment:
We need to sort account by their account number as string and not as number. 
```Ex. 1, 2, 20, 21, 3, 4, ....```
If we don't use cast, we have some things like : ```1, 2, 3, 4, 20, 21, ...```